sktime/performance_metrics/forecasting/probabilistic/_classes.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+
+import numpy as np
+
+# TODO: add formal tests
+import pandas as pd
+from sklearn.metrics._regression import _check_reg_targets
+
+from sktime.performance_metrics.forecasting._classes import _BaseForecastingErrorMetric
+
+
+def _check_y_pred_type(metric_object, y_pred):
```

Comment:
good idea - I guess this is where we would put checks and the conversions?

There's currently one conversion - `_convert_pred_interval_to_quantiles` in `forecasting.base._base`, and one check, `_check_predict_quantiles`. There is no check for the new "predict interval" format.

Reply:
Yes this is mostly a placeholder for now. I'll likely add to this in the next PR (adding interval metrics)